### PR TITLE
[gui] Fix snode offset (mesh disappearing bug)

### LINKED ISSUE #5571: [GGUI] Meshes disappear when using ti.arch=ti.vulkan
**Describe the bug**
After this pr #5430 , when running `ti example`(the 25th example), the cloth mesh disappears:

2022-07-29 16-30-40 的屏幕截图

And the right short cut should be like this:
2022-07-29 16-28-06 的屏幕截图

**To Reproduce**
Here is a test python script for you to try to fix this bug. Or you can run the 25th of taichi example.
```py
import taichi as ti

ti.init(arch=ti.vulkan)

N = 2
NV = (N + 1)**2
NT = 2 * N**2
NE = 2 * N * (N + 1) + N**2

pos = ti.Vector.field(3, ti.f32, shape=NV)
tri = ti.field(ti.i32, shape=3 * NT)
edge = ti.Vector.field(2, ti.i32, shape=NE)


@ti.kernel
def init_pos():
    for i, j in ti.ndrange(N + 1, N + 1):
        idx = i * (N + 1) + j
        pos[idx] = ti.Vector([i / N, 0.5, 1.0 - j / N])

@ti.kernel
def init_tri():
    for i, j in ti.ndrange(N, N):
        tri_idx = 6 * (i * N + j)
        pos_idx = i * (N + 1) + j
        if (i + j) % 2 == 0:
            tri[tri_idx + 0] = pos_idx
            tri[tri_idx + 1] = pos_idx + N + 2
            tri[tri_idx + 2] = pos_idx + 1
            tri[tri_idx + 3] = pos_idx
            tri[tri_idx + 4] = pos_idx + N + 1
            tri[tri_idx + 5] = pos_idx + N + 2
        else:
            tri[tri_idx + 0] = pos_idx
            tri[tri_idx + 1] = pos_idx + N + 1
            tri[tri_idx + 2] = pos_idx + 1
            tri[tri_idx + 3] = pos_idx + 1
            tri[tri_idx + 4] = pos_idx + N + 1
            tri[tri_idx + 5] = pos_idx + N + 2


@ti.kernel
def init_edge():
    for i, j in ti.ndrange(N + 1, N):
        edge_idx = i * N + j
        pos_idx = i * (N + 1) + j
        edge[edge_idx] = ti.Vector([pos_idx, pos_idx + 1])
    start = N * (N + 1)
    for i, j in ti.ndrange(N, N + 1):
        edge_idx = start + j * N + i
        pos_idx = i * (N + 1) + j
        edge[edge_idx] = ti.Vector([pos_idx, pos_idx + N + 1])
    start = 2 * N * (N + 1)
    for i, j in ti.ndrange(N, N):
        edge_idx = start + i * N + j
        pos_idx = i * (N + 1) + j
        if (i + j) % 2 == 0:
            edge[edge_idx] = ti.Vector([pos_idx, pos_idx + N + 2])
        else:
            edge[edge_idx] = ti.Vector([pos_idx + 1, pos_idx + N + 1])


init_pos()
init_tri()
init_edge()

window = ti.ui.Window("Display Mesh", (1024, 1024), vsync=True)
canvas = window.get_canvas()
scene = ti.ui.Scene()
camera = ti.ui.make_camera()
camera.position(-1.82731234,  2.26492691,  2.27800684)
camera.lookat(-1.14022911,  1.79402169,  1.72468514)
camera.fov(45)

while window.running:
    camera.track_user_inputs(window, movement_speed=0.003, hold_key=ti.ui.RMB)
    scene.set_camera(camera)
    scene.point_light(pos=(0.5, 1, 2), color=(1, 1, 1))
    
    scene.particles(pos, radius=0.01, color=(1.0, 1.0, 0.0))
    scene.mesh(pos, tri, color=(50/255, 144/255, 168/255))
    
    canvas.scene(scene)
    window.show()
    
```
**Bug Analysis**
The field memory layout based on ti.vulkan is changed, so when transfering verticles and indices data from field to vulkan buffer, it goes wrong.

May you help to solve this problem, thanks a lot!
cc @bobcao3

### PR DESCRIPTION
**Pull request by bobcao3**
Fixes #5571

### CODE REVIEW

**Review by Morcki — APPROVED** [review]
LGTM! Thanks a lot!

File: taichi/rhi/device.h [hunk 1]
```diff
@@ -199,7 +199,7 @@ enum class PolygonMode : int {
   Point = 2,
 };
 
-enum class TI_DLL_EXPORT BufferFormat : uint32_t {
```
Comment (ailzhang) [inline taichi/rhi/device.h:202]: @bobcao3 These were needed for texture aot, shall we keep them for now?

**Review by ailzhang — APPROVED** [review]
LGTM!

File: taichi/rhi/device.h [hunk 2]
```diff
@@ -199,7 +199,7 @@ enum class PolygonMode : int {
...
+enum class BufferFormat : uint32_t {
```
Comment (ailzhang) [inline taichi/rhi/device.h:202]:
```suggestion
enum class TI_DLL_EXPORT BufferFormat : uint32_t {
```

File: taichi/rhi/device.h [hunk 3]
```diff
@@ -254,9 +254,9 @@ class Pipeline {
   virtual ResourceBinder *resource_binder() = 0;
 };
 
-enum class TI_DLL_EXPORT ImageDimension { d1D, d2D, d3D };
+enum class ImageDimension { d1D, d2D, d3D };
```
Comment (ailzhang) [inline taichi/rhi/device.h:257]:
```suggestion
enum class TI_DLL_EXPORT ImageDimension { d1D, d2D, d3D };
```

File: taichi/rhi/device.h [hunk 4]
```diff
@@ -254,9 +254,9 @@ class Pipeline {
...
 
-enum class TI_DLL_EXPORT ImageLayout {
+enum class ImageLayout {
```
Comment (ailzhang) [inline taichi/rhi/device.h:259]:
```suggestion
enum class TI_DLL_EXPORT ImageLayout {
```

**Pull request merged**